src/sqlBible/src/SqlBible.ts

```diff
@@ -0,0 +1,292 @@
+        await this.dbReady;
+        return await getManager().save(section);
+    }
+
+    async addVersion(version: BibleVersion) {
+        await this.dbReady;
+        return getManager().save(version);
+    }
+
+    async createCrossReference(refRange: IBibleReferenceRangeVersion) {
+        return new BibleCrossReference(this.getNormalizedReference(refRange));
+    }
+
+    async generateBookMetadata(book: BibleBook) {
+        await this.dbReady;
+        const metaData = await getManager()
+            .createQueryBuilder(BiblePhrase, 'phrase')
+            .addSelect('COUNT(DISTINCT phrase.versionVerseNum)', 'numVerses')
+            .where({
+                id: Between(
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 1,
+                            normalizedVerseNum: 1,
+                            versionId: book.versionId
+                        },
+                        5
+                    ),
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 999,
+                            normalizedVerseNum: 999,
+                            versionId: book.versionId
+                        },
+                        5
+                    )
+                )
+            })
+            .orderBy('phrase.versionChapterNum')
+            .groupBy('phrase.versionChapterNum')
+            .getRawMany();
+        book.setChaptersMeta(metaData.map(chapterMetaDb => chapterMetaDb.numVerses));
+        return getManager().save(book);
+        console.log(metaData);
+    }
+
+    async getBookForVersionReference(reference: IBibleReferenceVersion) {
+        await this.dbReady;
+        return getManager().findOne(BibleBook, {
+            where: {
+                versionId: reference.versionId,
+                osisId: reference.bookOsisId
+            }
+        });
+    }
+
+    async getBooksForVersion(versionId: number) {
+        await this.dbReady;
+        return getManager().find(BibleBook, {
+            where: {
+                versionId
+            },
+            order: {
+                number: 'ASC'
+            }
+        });
+    }
+
+    async getNextPhraseIdForNormalizedVerseNum(
+        reference: IBibleReferenceNormalized,
+        versionId: number
+    ): Promise<number> {
+        await this.dbReady;
+        const lastPhrase = await getManager().find(BiblePhrase, {
+            where: {
+                id: Between(
+                    generateNormalizedRefId({ ...reference, versionId }, 5),
+                    generateNormalizedRefId(
+                        {
+                            ...reference,
+                            versionId,
+                            phraseNum: 99
+                        },
+                        5
```

Comment:
When I see a number `5` out of nowhere, I'm really confused. If it's necessary, use a descriptive enum

Reply:
refactored, see above